### PR TITLE
feat: add `array/base/cunone-by` package

### LINKED ISSUE #2325: [RFC]: add `@stdlib/array/base/cunone-by`
### Description

This RFC proposes adding the package `@stdlib/array/base/cunone-by`, which cumulatively tests whether no array element in a provided array passes a test implemented by a predicate function. The function should return a new generic array. The package should also provide an `#assign` API for setting output values in a provided output array.

```javascript
function isPositive( value ) {
	return ( value > 0 );
}

var x = [ 0, 0, 0, 1, 0 ];

var y1 = cunoneBy( x, isPositive );
// returns [ true, true, true, false, false ]

var y2 = [ false, null, false, null, false, null, false, null, false, null ];
var out = cunoneBy.assign( x, y2, 2, 0, isPositive );
// returns [ true, null, true, null, true, null, false, null, false, null ]

var bool = ( out === y2 );
// returns true
```

where the `assign` API supports an offset and stride (see, e.g., `@stdlib/array/base/take`).

Both APIs should support accessor arrays (see, e.g., `@stdlib/array/base/take`).

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/array/base/none-by`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Kaif987**
Resolves #2325

## Description

> What is the purpose of this pull request?

This pull request:

-   This pull request adds the `@stdlib/array/base/cunone-by` package that cumulatively tests whether no array element in a provided array passes a test implemented by a predicate function

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2325  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# cunone-by
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cunone-by/README.md]:
```suggestion
# cunoneBy
```

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+## Usage
+
+```javascript
+var cunoneBy = require( '@stdlib/array/base/cunone-by' );
+```
+
+#### cunoneBy( x, predicate\[, thisArg ] )
+
+Cumulatively test whether every array element in a provided array fails a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cunone-by/README.md]:
```suggestion
Cumulatively tests whether every array element in a provided array fails a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+var x = [ 0, 0, 0, 1, 0 ];
+
+var y = cunoneBy( x, fcn );
+// returns [ true, true, true, false, false ]
+```
+
+#### cunoneBy.assign( x, out, stride, offset, predicate\[, thisArg ] )
+
+Cumulatively test whether every array element in a provided array fails a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cunone-by/README.md]:
```suggestion
Cumulatively tests whether every array element in a provided array fails a test implemented by a predicate function and assigns the values to elements in a provided output array.
```

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    return ( value > 0 );
+}
+
+var x = [ 0, 0, 0, 1, 0 ];
+
+var y = cunoneBy( x, fcn );
+// returns [ true, true, true, false, false ]
+```
+
+#### cunoneBy.assign( x, out, stride, offset, predicate\[, thisArg ] )
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cunone-by/README.md]: We should move the documentation for the `assign` method to after L92.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Left a few comments concerning some doc fixes, but overall this looks very good!

**Comment by Kaif987** [comment]
Hey @Planeshifter I have made the necessary changes. Is there anything else that I need to add

File: lib/node_modules/@stdlib/array/base/cunone-by/README.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+var x = [ 0, 0, 0, 1, 0 ];
+
+var y = cunoneBy( x, fcn );
+// returns [ true, true, true, false, false ]
+```
+
+
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/cunone-by/README.md:50]:
```suggestion

```

**Review by Planeshifter — APPROVED** [review]
Let's land this! Thanks again for the PR!

**Pull request merged**